### DISCUSSION TITLE
Config to freeze mouse while scrolling windows

### DISCUSSION THREAD

**Discussion by Supreeeme** [YaLTeR/niri#613] [Ideas]
As an alternative to the scroll amount percentage config, I think it might be cool to have the mouse freeze in place while a window is in the process of moving/scrolling over. This would largely be useful for focus-follows-mouse, and would prevent issues of the wrong window getting focused while the correct one is moving into focus.

**Comment by YaLTeR** [comment]
Hm, when are you hitting a problematic case like this?

**Reply by Supreeeme** [reply]
Two monitors, current window is fullscreen, next window is fullscreen, try to grab the next window via struts, cursor might end up on the next monitor.

**Reply by YaLTeR** [reply]
So like, you move mouse towards the next monitor intending to hit the window in the strut, and then you want the cursor to freeze as soon as it activates focus-follow-mouse on that window so that it doesn't make it to the next monitor?

**Reply by Supreeeme** [reply]
Yep, that's the idea

**Reply by YaLTeR** [reply]
Could it work if you put the other screen a few logical px away from the main screen so the mouse can't go between them?

**Reply by Supreeeme** [reply]
So, I do want the ability to still go between the screens with the mouse. However, something interesting I've discovered is that even if there is a gap between the screens, I can still move the mouse between them. I can set the distance between the monitors to be about 200 px apart, and it will just require me to move the mouse faster to move it between the screens. I don't know if this is wanted (or even "fixable"?) behavior, but it works for me lol.

**Reply by YaLTeR** [reply]
Yeah, it's kind of semi-intended behavior, though not exactly reliable as this pixel distance will change with mouse Hz. (tbh it was just simpler to implement it this way)
